vnet/lib/vnet/models/route_link.rb

```diff
@@ -22,5 +22,24 @@ class RouteLink < Base
     translation_static_addresses: :destroy,
     _mac_address: :destroy
 
+    def self.lookup_by_nw(i_uuid, e_uuid)
+      i = lookup_nw(i_uuid)
+      e = lookup_nw(e_uuid)
+      self[:uuid => (i&e).first]
+    end
+
+    private
+
+    def self.lookup_nw(uuid)
+      RouteLink.dataset.join_table(
+        :inner,
+        :routes,
+        route_links__id: :routes__route_link_id
+      ).join_table(
+        :inner,
+        :networks,
+        routes__network_id: :networks__id
+      ).where(:networks__uuid => uuid.gsub("nw-","")).select(:route_links__uuid).all.map(&:uuid)
+    end
```

Comment:
https://github.com/axsh/openvnet/blob/develop/vnet/lib/vnet/models/ip_lease.rb#L57

The above is more like how it should look.